viabel/objectives.py

```diff
@@ -132,17 +130,33 @@ def __init__(self, approx, model, num_mc_samples, use_path_deriv=False, hessian_
                 'loo_diag_approx' : using "leave one out" method with hessian vector product value at other samples to
                     estimate the diagonal values of hessian
                 'loo_direct_approx;: the same method as 'loo_diag_approx' but use the scaled approximation to the
-                    grdient of scale to do the "loo" estimation
+                    gradient of scale to do the "loo" estimation
         """
         self._use_path_deriv = use_path_deriv
-        super().__init__(approx, model, num_mc_samples)
-
         self.hessian_approx_method = hessian_approx_method
+        super().__init__(approx, model, num_mc_samples)
 
     def _update_objective_and_grad(self):
         approx = self.approx
 
         def variational_objective(var_param):
```

Comment:
The `variational_objective` definition should be part of the `if self.hessian_approx_method is None:` logic since otherwise there's no need to define it